sops/__init__.py

```diff
@@ -1278,10 +1278,13 @@ def write_file(tree, path=None, filetype=None):
         fd = tempfile.NamedTemporaryFile(suffix="."+filetype, delete=False)
         path = fd.name
 
-    if fd and not isinstance(tree, dict) and not isinstance(tree, list):
```

Comment:
Probably but I was being consistent with the rest of this file. If we want to go the simpler route, I think it would be better suited in another PR as it might break something unexpectedly whereas this definitely only patches what is broken